### PR TITLE
Fix Mallet wrapper & test for HDPTransform

### PR DESCRIPTION
**Pull request by menshikh-iv**

### CODE REVIEW

File: gensim/models/wrappers/ldamallet.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def __getitem__(self, bow, iterations=100):
 
     def load_word_topics(self):
         logger.info("loading assigned topics from %s", self.fstate())
-        word_topics = numpy.zeros((self.num_topics, self.num_terms), dtype=numpy.float32)
+        word_topics = numpy.zeros((self.num_topics, self.num_terms), dtype=numpy.float64)
```
Comment (piskvorky) [inline gensim/models/wrappers/ldamallet.py]: Is this really necessary? Double precision is rarely useful.

Comment (menshikh-iv) [reply gensim/models/wrappers/ldamallet.py]: Double precision used for all TMs now.

Comment (piskvorky) [reply gensim/models/wrappers/ldamallet.py]: Question is "why"?

Comment (menshikh-iv) [reply gensim/models/wrappers/ldamallet.py]: I don't know, I don't implement TMs in gensim.

Comment (piskvorky) [reply gensim/models/wrappers/ldamallet.py]: That doesn't sound right. I'm pretty sure many models used to use single precision (RP and LSI, for sure).

Can you check when the change to double happened, and why?

**Pull request merged**